bigbluebutton/greenlight
# feat: manage a global duration of sessions

## Description

This feature provide a way to set a global duration for all bbb room start.
We can define 3 variables:
* `DEFAULT_MEETING_DURATION`: the default duration of all room - 0 means unset
* `DEFAULT_CONFIGURABLE_MIN_MEETING_DURATION`: the minimum duration that can be set from the admin UI
* `DEFAULT_CONFIGURABLE_MAX_MEETING_DURATION`: the maximum duration that can be set from the admin UI.
These variables can be set from environment, and the `DEFAULT_MEETING_DURATION` can be overriden into the admin room configuration.

Also show on main page of user rooms the duration set by the admin, if the setting is set to a value upper to 0.

A topic on mailing list was open [HERE](https://groups.google.com/d/msgid/bigbluebutton-greenlight/ceecf493-49cc-0715-1308-9ad08d68b7aa%40recia.fr?utm_medium=email&utm_source=footer)
## Testing Steps

set the variables from ENVIRONMENT and also test to set the duration on the admin room configuration site and start a room. Check the param provided to BBB.

## Screenshots (if appropriate):

### Admin room configuration vue
![Sélection_206](https://user-images.githubusercontent.com/1597283/131529574-eda8a5b9-04b1-45bd-b3c3-65d88615fc8a.png)
### User main room view
![Sélection_207](https://user-images.githubusercontent.com/1597283/131550087-235cb97b-201c-4ac4-8f45-8bf550a2dcf1.png)
